### ISSUE TITLE
Error and warning messages with clojure.string/replace

### ISSUE THREAD

**Issue by Asketre** [clj-kondo/clj-kondo#2555]
**platform**

MacOS JVM

**editor**

Latest version of Visual Code / Calva 

**problem**

false-positive warning and error messages related to clojure.string/replace
Maybe linked to closed issue #2415

**repro**

See files sample1 and sample2 in  [Sample Code](https://github.com/Asketre/sample)

and screenshots below

![Image](https://github.com/user-attachments/assets/e07b6e26-6cf0-47bd-9dd2-82e40e49d85c)
![Image](https://github.com/user-attachments/assets/9b848400-1ce6-493f-88c2-f93db66161fa)

Thank you !

**Comment by borkdude** [comment]
Hey @Asketre, thanks for posting the issue. The first one I consider a valid warning. You should require namespace before usage even if Clojure might have already loaded them first (which is an implementation detail).

The second one is surely a false positive, thanks.

A shorter repo that I can use as a test:

```
$ clj-kondo --lint - <<< "(require '[clojure.string :as str]) (defn replace-str [_foo bar] bar) (str/replace \"foo\" #\"bar\" (partial replace-str \"dude\"))"
<stdin>:1:97: warning: Regex match arg requires string or function replacement arg.
linting took 5ms, errors: 0, warnings: 1
```

**Comment by Asketre** [comment]
Thank you very much :)
